File: Tan Khoa/24strb3.py
n = int(input())
price = list(map(int, input().split()))

price.sort(reverse = True)

max_revenue, min_price = 0, int(1e9)
for i in range(n):
	if price[i]*(i+1) >= max_revenue:
		max_revenue = price[i]*(i+1)
		min_price = price[i]

print(max_revenue, min_price)

File: Tan Khoa/24strb3.cpp
#include <bits/stdc++.h>

using namespace std;

#define nln '\n'
typedef long long ll;

int main() {
	cin.tie(0)->sync_with_stdio(0);
	ll n;	cin >> n;
	vector<ll> arr(n);
	for (auto &v : arr)
		cin >> v;
	sort(arr.begin(), arr.end());

	ll min_price = LLONG_MAX, max_revenue = 0;
	for (int i = 0; i < n; ++i) {
		if (arr[i] * (n-i) > max_revenue) {
			max_revenue = arr[i] * (n-i);
			min_price = arr[i];
		}
	}

	cout << max_revenue << ' ' << min_price << endl;

	return 0;
}